certificate-service/app/src/main/java/se/inera/intyg/certificateservice/application/certificate/service/CertificateDataValueConverter.java

```diff
@@ -7,22 +7,21 @@
 import se.inera.intyg.certificateservice.domain.certificate.model.ElementValueDate;
 import se.inera.intyg.certificateservice.domain.certificatemodel.model.ElementConfigurationDate;
 import se.inera.intyg.certificateservice.domain.certificatemodel.model.ElementSpecification;
+import se.inera.intyg.certificateservice.domain.certificatemodel.model.ElementType;
 
 @Component
 public class CertificateDataValueConverter {
 
   public CertificateDataValue convert(ElementSpecification elementSpecification,
       ElementValue elementValue) {
-    return switch (elementSpecification.configuration().type()) {
-      case CATEGORY -> null;
-      case DATE -> {
-        final var value = elementValue != null ? ((ElementValueDate) elementValue).date() : null;
-        final var configuration = (ElementConfigurationDate) elementSpecification.configuration();
-        yield CertificateDataValueDate.builder()
-            .id(configuration.id())
-            .date(value)
-            .build();
-      }
-    };
+    if (ElementType.DATE.equals(elementSpecification.configuration().type())) {
+      final var value = elementValue != null ? ((ElementValueDate) elementValue).date() : null;
+      final var configuration = (ElementConfigurationDate) elementSpecification.configuration();
+      return CertificateDataValueDate.builder()
+          .id(configuration.id())
+          .date(value)
+          .build();
+    }
+    return null;
```

Comment:
If it reaches this I think it is better that we throw an exception than return null. For example, IllegalStateException which a message saying what type of configuration that didn't match an alternative. If it comes to this we have a bug in the code before and it is good to fail fast here.